go/vt/topo/tablet.go

```diff
@@ -263,13 +267,21 @@ func (ts *Server) GetTabletsByCell(ctx context.Context, cellAlias string, opt *G
 		return nil, err
 	}
 
-	tablets := make([]*TabletInfo, len(listResults))
+	tablets := make([]*TabletInfo, 0)
```

Comment:
Nit, but we can keep the older optimization, using the list size for cap rather than len, while avoiding over allocating in the new path:
```
var sizeHint int
if opt.Keyspace == nil && opt.Shard == nil {
    sizeHint = len(listResults)
}
tablets := make([]*TabletInfo, 0, sizeHint)
```

Reply:
Oh, and opt is a pointer. So we should check for nil when looking at its members as well.

Reply:
Updated